tmt/steps/prepare/feature/fips-enable.yaml

```diff
@@ -0,0 +1,196 @@
+---
+- name: Enable FIPS mode on RHEL-7, RHEL-8, RHEL-9 and RHEL-10
+  hosts: all
+
+  tasks:
+    - name: Fail on distribution other than RHEL or CentOS
+      ansible.builtin.fail:
+        msg: "We support FIPS mode on RHEL or CentOS, only!"
+      when:
+        - ansible_facts['distribution'] != 'RedHat'
+        - ansible_facts['distribution'] != 'CentOS'
+
+    - name: Check if SUT is a container
+      ansible.builtin.command: cat /proc/1/environ
+      register: proc_1_environ
+      changed_when: false
+
+    - name: Register cpuinfo
+      ansible.builtin.command: cat /proc/cpuinfo
+      register: cpuinfo_contents
+      changed_when: false
+
+    - name: Enforce modulus bits for OpenSSL
+      ansible.builtin.shell: |
+        changed=0
+        if ! grep 'OPENSSL_ENFORCE_MODULUS_BITS' /etc/environment; then
+            echo 'OPENSSL_ENFORCE_MODULUS_BITS=true' >> /etc/environment
+            changed=1
+        fi
+        if [ -f /etc/profile.d/openssl.sh ] || \
+          ! grep 'export OPENSSL_ENFORCE_MODULUS_BITS=true' /etc/profile.d/openssl.sh; then
+          echo 'export OPENSSL_ENFORCE_MODULUS_BITS=true' > /etc/profile.d/openssl.sh
+          chmod +x /etc/profile.d/openssl.sh
+          changed=1
+        fi
+        if [ -f /etc/profile.d/openssl.csh ] || \
+          ! grep 'setenv OPENSSL_ENFORCE_MODULUS_BITS true' /etc/profile.d/openssl.csh; then
+          echo 'setenv OPENSSL_ENFORCE_MODULUS_BITS true' > /etc/profile.d/openssl.csh
+          chmod +x /etc/profile.d/openssl.csh
+          changed=1
+        fi
+        test "$changed" -eq 1
+      register: output
+      changed_when: output.rc == 0
+
+    - name: Disable prelink
+      ansible.builtin.shell: |
+        if rpm -q prelink; then
+          if pgrep prelink; then
+            killall prelink && echo "CHANGED"
+          fi
+          if ! grep "PRELINKING=no" /etc/sysconfig/prelink; then
+            sed -i 's/PRELINKING=.*/PRELINKING=no/g' /etc/sysconfig/prelink
+            prelink -u -a
+            killall prelink
+            echo "CHANGED"
+          fi
+        fi
+      register: output
+      changed_when: '"CHANGED" in output.stdout'
+      when: ansible_facts['distribution_major_version'] | int == 7
+
+    - name: Install crypto-policies-scripts and dracut-fips
+      ansible.builtin.dnf:
+        name:
+          - crypto-policies-scripts
+          - dracut-fips
+        state: present
+      when: ansible_facts['distribution_major_version'] | int > 7
+
+    # skip_ansible_lint is present int the following two tasks to prevent
```

Comment:
```suggestion
    # skip_ansible_lint is present in the following two tasks to prevent
```

Reply:
@The-Mule couldn't we use the `package` module? Or does that also have issues?

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html